src/Processors/QueryPlan/Optimizations/optimizeUseNormalProjection.cpp

```diff
@@ -187,7 +189,13 @@ std::optional<String> optimizeUseNormalProjections(Stack & stack, QueryPlan::Nod
         if (!analyzed)
             continue;
 
-        if (candidate.sum_marks >= ordinary_reading_marks)
+        if (candidate.sum_marks > ordinary_reading_marks)
+            continue;
+
+        /// Both optimize_use_projections and force_optimize_projection are true, consider projection when equal cost.
+        if (candidate.sum_marks == ordinary_reading_marks
```

Comment:
@jsc0218 thanks for working on this fix.
Asking some questions for my understanding.
On our original issue (https://github.com/ClickHouse/ClickHouse/issues/73243), the number of granules reads was not equal for candidate (projection) vs ordinary (table):
- Using projection: `Granules: 1/123`
- Using table: `Granules: 123/123`

Is the number of granules = to sum marks? 
Shouldn't `candidate.sum_marks` be < `ordinary_reading_marks` for the reproducible case?

Reply:
Sorry for replying late as I was on vacation, and good question, the issue you mentioned actually needs further investigation. This pr fixed the case of `SELECT * FROM test /*WHERE user_name = 'John#999'*/ SETTINGS optimize_use_projections=1, force_optimize_projection=1 ORDER BY user_name, date;`